jte/src/test/java/gg/jte/TemplateEngineTest.java

```diff
@@ -543,6 +543,132 @@ void tagWithVarArgs4() {
         thenOutputIs("test.key with Hello World ");
     }
 
+    @Test
+    void templateCall() {
+        dummyCodeResolver.givenCode("module/components/card.jte", "@param java.lang.String firstParam\n" +
+                "@param int secondParam\n" +
+                "One: ${firstParam}, two: ${secondParam}");
+        givenTemplate("@module.components.card(model.hello, model.x), That was an arbitrary template call!");
+        thenOutputIs("One: Hello, two: 42, That was an arbitrary template call!");
+    }
+
+    @Test
+    void templateCall_noConflictWithKeywords_for() {
+        dummyCodeResolver.givenCode("format.jte", "@param java.lang.String firstParam\n" +
+                "@param int secondParam\n" +
+                "One: ${firstParam}, two: ${secondParam}");
+        givenTemplate("@format(model.hello, model.x), That was an arbitrary template call!");
+        thenOutputIs("One: Hello, two: 42, That was an arbitrary template call!");
+    }
+
+    @Test
+    void templateCall_noConflictWithKeywords_param() {
+        dummyCodeResolver.givenCode("parameter.jte", "@param java.lang.String firstParam\n" +
+                "@param int secondParam\n" +
+                "One: ${firstParam}, two: ${secondParam}");
+        givenTemplate("@parameter(model.hello, model.x), That was an arbitrary template call!");
+        thenOutputIs("One: Hello, two: 42, That was an arbitrary template call!");
+    }
+
+    @Test
+    void templateCall_noConflictWithKeywords_import() {
+        dummyCodeResolver.givenCode("importer.jte", "@param java.lang.String firstParam\n" +
+                "@param int secondParam\n" +
+                "One: ${firstParam}, two: ${secondParam}");
+        givenTemplate("@importer(model.hello, model.x), That was an arbitrary template call!");
+        thenOutputIs("One: Hello, two: 42, That was an arbitrary template call!");
+    }
+
+    @Test
+    void templateCall_singleAt() {
+        givenTemplate("@");
+        thenOutputIs("@");
+    }
+
+    @Test
+    void templateCall_singleAt_ws() {
+        givenTemplate("@\nfoo");
+        thenOutputIs("@\nfoo");
+    }
+
+    @Test
+    void escapeAt_templateCall() {
+        givenTemplate("mail@@example.com(write me)"); // Escaped with double @
+        thenOutputIs("mail@example.com(write me)");
+    }
+
+    @Test
+    void escapeAt_3x() {
+        givenTemplate("@@@");
+        thenOutputIs("@@");
```

Comment:
This should throw a syntax error IMO, so all `@` characters are reserved and have to be escaped or might get used in keywords in later releases. That way, there won't be BC breaks if new keywords starting with `@` are introduced.

Reply:
With the new keyword we don't absolutely need escaping for this PR, so I removed it for now. You're right, and the current implementation is a little bit too rushed to be any good :-D.

Will continue this in https://github.com/casid/jte/issues/125